File: web-app/src/app/core/models/limpieza.ts
export interface LimpiezaResponse {
  id: string;
  habitacionId: string;
  habitacionNumero: string;
  usuarioId: string;
  usuarioNombre: string;
  fechaInicio: string;
  fechaFin: string | null;
  duracionSegundos: number | null;
  createdAt: string;
}

export interface IniciarLimpiezaRequest {
  habitacionId: string;
  usuarioId: string;
}

File: api-rest/hotel_cervera_api/src/main/java/com/hotel/cervera/hotel_cervera_api/model/Limpieza.java
package com.hotel.cervera.hotel_cervera_api.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "limpiezas")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Limpieza {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "habitacion_id", nullable = false)
    private Habitacion habitacion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @Column(name = "fecha_inicio", nullable = false)
    private OffsetDateTime fechaInicio;

    @Column(name = "fecha_fin")
    private OffsetDateTime fechaFin;

    @Column(name = "duracion_segundos")
    private Integer duracionSegundos;

    @CreationTimestamp
    @Column(name = "fecha_creacion", updatable = false)
    private OffsetDateTime fechaCreacion;
}
